### PR TITLE
Fix post likes endpoint

### LINKED ISSUE #210: /api/author/{author_id}/posts/{posts_id}/likes not matching spec
Endpoint not returning data in a format that matches the spec.

### PR DESCRIPTION
**Pull request by olivier-vadiaval**
Fix the endpoint for post likes
Add tests for the post likes endpoint
Add 2 more tests for the comment likes endpoint

Post likes endpoint was returning post authors instead of likes objects

Resolves #210 

If you run all the unit tests, please note that the tests to PUT and POST posts (three tests in total) don't seem to pass for me
I tried upgrading my python version but still getting the errors
I saw the comments in Tyler's pr related to that
So, I'm unsure if it's actually related to the unit test code itself or if it's a dev environment issue on my end.

That aside, the unit tests in test_likes.py should pass.

### CODE REVIEW

**Review by tylerelston — APPROVED** [review]
Thanks! I'm not getting any failing tests when I run them so it may be deve env related? Also, does this resolve #198 ?

File: code/api/views.py [hunk 1]
```diff
@@ -425,14 +425,35 @@ def get(self, request, author_id, post_id):
...
-            post = LocalPost.objects.get(id=post_id)
-            authors = [like.author.as_json() for like in post.likes.all()]
+            author = get_object_or_404(Author, id=author_id)
+            post = get_object_or_404(LocalPost, id=post_id, author=author)
+
+            post_likes = post.likes.all()
+
+            items = []
+            for like in post_likes:
+                if Author.objects.filter(url=like.author.get_url_id()):
```
Comment (debecerra) [inline code/api/views.py:435]: Not sure what this if clause is doing? Doesn't it always return true?

Comment (debecerra) [reply code/api/views.py:435]: You don't have `exists()` and regardless I'm not sure that you need it

**Review by debecerra — APPROVED** [review]
Tests all pass for me as well. Also checked our frontend for liking posts and appears to be working as expected.

File: code/api/views.py (hunk 1 above)
Comment (olivier-vadiaval) [reply code/api/views.py:435]: You're right, it should actually have exist

Comment (olivier-vadiaval) [reply code/api/views.py:435]: Actually, yeah, this shouldn't be here

**Pull request merged**